api/healthcheck/composite_indicator.go

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2018 The Service Manager Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package healthcheck
+
+import "github.com/Peripli/service-manager/pkg/health"
+
+// compositeIndicator aggregates multiple health indicators and provides one detailed health
+type compositeIndicator struct {
+	aggregationPolicy health.AggregationPolicy
+	indicators        []health.Indicator
+}
+
+// newCompositeIndicator returns a new compositeIndicator for the provided health indicators
+func newCompositeIndicator(indicators []health.Indicator, aggregator health.AggregationPolicy) health.Indicator {
+	return &compositeIndicator{
+		aggregationPolicy: aggregator,
+		indicators:        indicators,
+	}
+}
+
+// Name returns the name of the compositeIndicator
+func (i *compositeIndicator) Name() string {
+	return "CompositeHealthIndicator"
```

Comment:
I guess it would be more useful if the name of the internal indictator is actually composed on the names of the indicators slice. (especially since we now log it in the Controller)